sql/src/test/java/org/apache/druid/sql/calcite/CalciteSubqueryTest.java

```diff
@@ -675,98 +678,112 @@ public void testEmptyGroupWithOffsetDoesntInfiniteLoop()
   public void testMaxSubqueryRows()
   {
     if ("without memory limit".equals(testName)) {
-      expectedException.expect(ResourceLimitExceededException.class);
-      expectedException.expectMessage(
-          "Cannot issue the query, subqueries generated results beyond maximum[1] rows. Try setting the "
-          + "'maxSubqueryBytes' in the query context to 'auto' for enabling byte based limit, which chooses an optimal "
-          + "limit based on memory size and result's heap usage or manually configure the values of either 'maxSubqueryBytes' "
-          + "or 'maxSubqueryRows' in the query context. Manually alter the value carefully as it can cause the broker to "
-          + "go out of memory."
-      );
+      testMaxSubqueryRowsWithoutMemoryLimit();
+    } else {
+      testMaxSubQueryRowsWithLimit();
+    }
+  }
+
+  private void testMaxSubqueryRowsWithoutMemoryLimit()
+  {
+    Throwable exception = assertThrows(ResourceLimitExceededException.class, () -> {
       Map<String, Object> modifiedQueryContext = new HashMap<>(queryContext);
```

Comment:
It would be better to move all the test setup outside the `assertThrows` and narrow it down to the thing that we're expecting to throw an exception.